src/qt/walletmodel.cpp

```diff
@@ -756,3 +779,296 @@ bool WalletModel::getDefaultWalletRbf() const
+            ++i;
+            continue;
+        }
+
+        const int confirms = val.get_int ();
+        LogPrintf ("Pending Name FirstUpdate Confirms: %d\n", confirms);
+
+        if ( confirms < 12)
+        {
+            ++i;
+            continue;
+        }
+
+        if(getEncryptionStatus() == Locked)
+        {
+            if (QMessageBox::Yes != QMessageBox::question(this,
+                  tr("Confirm wallet unlock"),
+                  tr("Namecoin Core is about to finalize your name registration for name <b>%1</b>, by sending name_firstupdate. If your wallet is locked, you will be prompted to unlock it. Pressing cancel will delay your name registration by one block, at which point you will be prompted again. Would you like to proceed?").arg(QString::fromStdString(name)),
+                  QMessageBox::Yes|QMessageBox::Cancel,
+                  QMessageBox::Cancel))
+            {
+                LogPrintf ("User cancelled wallet unlock pre-name_firstupdate. Waiting 1 block.\n");
+                return;
+            }
+
+            LogPrintf ("Attempting wallet unlock ...\n");
+            WalletModel::UnlockContext ctx(this->requestUnlock ());
+            if (!ctx.isValid ())
+            {
+                return;
+            }
+            else
+            {
+                // NOTE: the reason we're doing this here and not at the same
+                // time as the one beneath it is because when ctx is destroyed
+                // (subsequently after leaving this block) the wallet is locked.
+                completedResult = this->completePendingNameFirstUpdate(name, rand, txid, data, toaddress);
+            }
+        }
+        else
+        {
+            completedResult = this->completePendingNameFirstUpdate(name, rand, txid, data, toaddress);
+        }
+
+        // if we got an error on name_firstupdate. prompt user for what to do
+        if(!completedResult.empty())
+        {
+            QString errorMsg = tr("Namecoin Core has encountered an error while attempting to complete your name registration for name <b>%1</b>. The name_firstupdate operation caused the following error to occurr:<br><br>%2<br><br>Would you like to cancel the pending name registration?")
+                .arg(QString::fromStdString(name))
+                .arg(QString::fromStdString(completedResult));
+            // if they didnt hit yes, move onto next pending op, otherwise
+            // the pending transaction will be deleted in the subsequent block
+            if (QMessageBox::Yes != QMessageBox::question(this, tr("Name registration error"),
+                                                          errorMsg,
+                                                          QMessageBox::Yes|QMessageBox::No,
+                                                          QMessageBox::No))
+            {
+                ++i;
+                continue;
+            }
+        }
+
+        pendingNameFirstUpdate.erase(i++);
```

Comment:
I think that the code would be cleaner as a for-each loop, and at least without the explicit `++i` (i. e., with `++i` in the loop).  I understand that this won't work since you erase iterators on the fly while looping -- but you could build up an array of things to remove during the iteration, and then just remove all of that in a follow-up step.  IMHO that would make the code clearer to read, and also less susceptible to bugs like the one that was here before.